### PR TITLE
Create Gold.java

### PR DESCRIPTION
**Pull request by gourabk121**
goldmine

Fixes #1073 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Goldmine problem

#### Languages Used:
- java

#### Files Added:
- Gold.java

### CODE REVIEW

**Comment by somya-kapoor** [comment]
Also please remove extra bullets points.

File: Goldmine/Gold.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Girlscript Summer Of Code
+// goldmine problem  in java
+
+
+//Given a gold mine of a*b dimensions.
```
Comment (KanchanThareja) [inline Goldmine/Gold.java]: Remove line 3

File: Goldmine/Gold.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+//Each field in this mine contains a positive integer which is the amount of gold in tons. 
+//Initially the miner is at first column but can be at any row.
+//He can move only (right->,right up /,right down) that is from a given cell,
+//the miner can move to the cell diagonally up towards the right or right or diagonally down towards the right.
+//Find out maximum amount of gold he can collect.
+
+
+import java.util.*;
```
Comment (KanchanThareja) [inline Goldmine/Gold.java]: remopve line 11

File: Goldmine/Gold.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+               arr[i][j] = scan.nextInt();
+            }
+       }
+       System.out.print(getMaxGold(arr, m, n)); 
+    } 
+} 
+
+
+//Input : mat[][] = {{5,4,2},
+//                   {1,4,2},
```
Comment (KanchanThareja) [inline Goldmine/Gold.java]: remove line 60

File: Goldmine/Gold.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                int right_up = (row == 0 ||col == n-1) ? 0 :goldTable[row-1][col+1];
+                int right_down = (row == m-1|| col == n-1) ? 0 :goldTable[row+1][col+1]; 
+                goldTable[row][col] = gold[row][col]+ Math.max(right, Math.max(right_up,right_down)); 
+            } 
+        } 
+        int res = goldTable[0][0];
+        for (int i = 1; i < m; i++) 
+            res = Math.max(res, goldTable[i][0]); 
+        return res; 
+    }
```
Comment (KanchanThareja) [inline Goldmine/Gold.java]: add comments to this class as well

File: Goldmine/Gold.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+       for(i=0; i<m; i++)
+       {    
+            for(j=0; j<n; j++)
+            {
+               arr[i][j] = scan.nextInt();
+            }
+       }
+       System.out.print(getMaxGold(arr, m, n)); 
+    } 
+}
```
Comment (KanchanThareja) [inline Goldmine/Gold.java]: fix indentation here

File: Goldmine/Gold.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+//the miner can move to the cell diagonally up towards the right or right or diagonally down towards the right.
+//Find out maximum amount of gold he can collect.
+
+
+import java.util.*; 
+  
+class Hello
+{ 
+    static final int MAX = 100;
+    static int getMaxGold(int gold[][],int m, int n){
```
Comment (somya-kapoor) [inline Goldmine/Gold.java]: Please add space after gold[][] and m

File: Goldmine/Gold.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        } 
+        int res = goldTable[0][0];
+        for (int i = 1; i < m; i++) 
+            res = Math.max(res, goldTable[i][0]); 
+        return res; 
+    } 
+    
+    public static void main(String arg[]) 
+    { 
+       int m,n, i, j;
```
Comment (somya-kapoor) [inline Goldmine/Gold.java]: add space before n

File: Goldmine/Gold.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+       int arr[][] = new int[10][10];
+       Scanner scan = new Scanner(System.in);
+       // enter row and column for array.
+       System.out.print("Enter row for the array (max 10) : ");
+       m = scan.nextInt();
+       System.out.print("Enter column for the array (max 10) : ");
+       n = scan.nextInt();
+       // enter array elements.
+       System.out.println("Enter " +(m*n)+ " Array Elements : ");
+       for(i=0; i<m; i++)
```
Comment (somya-kapoor) [inline Goldmine/Gold.java]: Add space around =,<

**Comment by KanchanThareja** [comment]
@gourabk121 code looks good just fix the pr description.

**Comment by gourabk121** [comment]
@KanchanThareja  is that fine now? I commited pr description also

**Comment by KanchanThareja** [comment]
hey @gourabk121 PR description means your very first comment on this page.

**Comment by gourabk121** [comment]
@KanchanThareja  I changed that also

**Comment by KanchanThareja** [comment]
@gourabk121 remove extra bullet points and fill in what are the changes proposed in the pr.

**Comment by gourabk121** [comment]
@KanchanThareja  is that ok now?

**Comment by KanchanThareja** [comment]
@gourabk121 good work
@jainaman224 I approve this PR. Please take a final look.

**Pull request merged**